extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/IntervalUtils.java

```diff
@@ -61,4 +64,20 @@ public static List<Interval> difference(final List<Interval> list1, final List<I
 
     return retVal;
   }
+
+  /**
```

Comment:
Maybe add this method to `Intervals` class instead, as it could have uses outside the MSQ extension too.

Reply:
Left this method in the interval utils since the top-level Javadoc for IntervalUtils mentions that the methods are specific to MSQ for now, and would make more sense to move to Intervals if MSQ wasn't an extension